app/controllers/admin/volunteers.py

```diff
@@ -67,33 +73,60 @@ def updateVolunteerTable(eventID):
         abort(404)
 
     program = event.singleProgram
-    # TODO: What do we do for no programs or multiple programs?
-    if not program:
-        return "TODO: What do we do for no programs or multiple programs?"
-
+    
     volunteerUpdated = updateEventParticipants(request.form)
     if volunteerUpdated:
         flash("Volunteer table succesfully updated", "success")
     else:
         flash("Error adding volunteer", "danger")
     return redirect(url_for("admin.trackVolunteersPage", eventID=eventID))
 
-@admin_bp.route('/addVolunteerToEvent/<volunteer>/<eventId>', methods = ['POST'])
-def addVolunteer(volunteer, eventId):
-    username = volunteer.strip("()").split('(')[-1]
-    user = User.get(User.username==username)
-    successfullyAddedVolunteer = addVolunteerToEventRsvp(user, eventId)
-    EventParticipant.create(user=user, event=eventId) # user is present
-    if successfullyAddedVolunteer:
-        flash("Volunteer successfully added!", "success")
+@admin_bp.route('/addVolunteersToEvent/<eventId>', methods = ['POST'])
+@admin_bp.route('/addVolunteerToEvent/<eventId>/<volunteer>', methods = ['POST'])
+def addVolunteer(eventId, volunteer = None):
+    successfullyAddedVolunteer = False
+    usernameList = []
+    if volunteer == None:
+        usernameList = request.form.getlist("volunteer[]")
     else:
-        flash("Error when adding volunteer", "danger")
-    return ""
+        username = volunteer.strip("()").split('(')[-1]
+        user = User.get(User.username==username)
+        usernameList.append(user)
+    eventParticipants = getEventParticipants(eventId)
+    for user in usernameList:
+        user = User.get(User.username==user)
+
+        if len(usernameList) == 0:
+            noVolunteersSelectedMessage = "No volunteers selected, please select a volunteer."
+            successfullyAddedVolunteer = False
+            noVolunteerSelected = True
+
+        if len(eventParticipants) == 0:
+            addVolunteerToEventRsvp(user, eventId)
+            EventParticipant.create(user = user, event = eventId)
+            successfullyAddedVolunteer = True
+
+        elif (EventRsvp.select().where(EventRsvp.user==user, EventRsvp.event_id == eventId) and
+              EventParticipant.select().where(EventParticipant.user == user, EventParticipant.event_id == eventId)):
+            duplicateVolunteerMessage = "This volunteer has already been added to this event."
+            successfullyAddedVolunteer = False
```

Comment:
I think it should still be successfully added. The user wants to add a volunteer, so if they are already in the db, we can consider that a success. It does mean something weird is going on, perhaps, but the end result is still a success.